cpp/src/arrow/filesystem/filesystem.h

```diff
@@ -325,9 +337,13 @@ class ARROW_EXPORT SlowFileSystem : public FileSystem {
 /// \param[out] out_fs FileSystem instance.
 /// \param[out] out_path (optional) Path inside the filesystem.
 /// \return Status
-ARROW_EXPORT
-Status FileSystemFromUri(const std::string& uri, std::shared_ptr<FileSystem>* out_fs,
-                         std::string* out_path = NULLPTR);
+ARROW_DEPRECATED("Use Result-returning version")
+inline Status FileSystemFromUri(const std::string& uri,
+                                std::shared_ptr<FileSystem>* out_fs,
+                                std::string* out_path = NULLPTR) {
+  ARROW_ASSIGN_OR_RAISE(*out_fs, FileSystemFromUri(uri, out_path));
```

Comment:
You can simply write `return FileSystemFromUri(uri, out_path).Value(out_fs);`

Reply:
Oops, I missed that feature!